src/clandinin_lab_to_nwb/brezovec/brezovecimagingextractor.py

```diff
@@ -0,0 +1,237 @@
+"""Specialized extractor for reading TIFF files produced via ScanImage.
+
+Classes
+-------
+ScanImageTiffImagingExtractor
+    Specialized extractor for reading TIFF files produced via ScanImage.
+"""
+import logging
+import re
+from collections import Counter
+from itertools import islice
+from pathlib import Path
+from types import ModuleType
+from typing import Optional, Tuple, Union, List, Dict
+from xml.etree import ElementTree
+
+import numpy as np
+from roiextractors.multiimagingextractor import MultiImagingExtractor
+from roiextractors.imagingextractor import ImagingExtractor
+from roiextractors.extraction_tools import PathType, get_package, DtypeType, ArrayType, FloatType
+from neuroconv.utils import calculate_regular_series_rate
+
+
+def _get_nifti_reader() -> ModuleType:
+    return get_package(package_name="nibabel", installation_instructions="pip install nibabel")
+
+
+def _determine_imaging_is_volumetric(folder_path: PathType) -> bool:
+    """
+    Determines whether imaging is volumetric based on 'zDevice' configuration value.
+    The value is expected to be '1' for volumetric and '0' for single plane images.
+    """
+    xml_root = _parse_xml(folder_path=folder_path)
```

Comment:
The parsing is duplicated here. Maybe this function could be converted in a simliar way to:

https://github.com/catalystneuro/clandinin-lab-to-nwb/pull/5

Or probably easier just make it a class method and call it after the parsing in line 108 to avoid the double parsing.

Reply:
Done as you suggested, made it a class method and called after parsing